### PR TITLE
Associate current cycle to nodeDef on creation

### PR DESCRIPTION
**Pull request by fefrain**

### CODE REVIEW

File: test/it/surveyTests/nodeDefTest.js [hunk 1]
```diff
@@ -12,9 +12,21 @@ const fetchRootNodeDef = async () => {
   return await NodeDefRepository.fetchRootNodeDef(Survey.getId(survey), true)
 }
 
-const createNodeDef = async (parentNodeUuid, type, name) => {
+function createNodeDef (parentNodeUuid, type, name) {
```
Comment (minotogna) [inline test/it/surveyTests/nodeDefTest.js]: why not ```const createNodeDef```?

File: webapp/survey/nodeDefs/actions.js [hunk 2]
```diff
@@ -56,14 +57,13 @@ const _updateParentLayout = (nodeDef, deleted = false) => async (dispatch, getSt
...
 
 export const createNodeDef = (parentUuid, type, props) => async (dispatch, getState) => {
-  const survey = SurveyState.getSurvey(getState())
-  const surveyId = Survey.getId(survey)
-  const surveyInfo = Survey.getSurveyInfo(survey)
-  const lastSurveyCycle = R.pipe(Survey.getCycles, R.keys, R.head)(surveyInfo)
+  const state = getState()
+  const surveyId = SurveyState.getSurveyId(state)
+  const user = AppState.getUser(state)
+  const cycle = User.getPrefSurveyCycle(surveyId)(user)
```
Comment (minotogna) [inline webapp/survey/nodeDefs/actions.js]: you should use ```SurveyState.getSurveyCycleKey```

File: test/it/surveyTests/nodeDefTest.js (hunk 1 above)
Comment (fefrain) [reply test/it/surveyTests/nodeDefTest.js]: fixed

File: webapp/survey/nodeDefs/actions.js (hunk 2 above)
Comment (fefrain) [reply webapp/survey/nodeDefs/actions.js]: ok

File: webapp/survey/nodeDefs/actions.js [hunk 3]
```diff
@@ -56,14 +57,12 @@ const _updateParentLayout = (nodeDef, deleted = false) => async (dispatch, getSt
...
+  const state = getState()
+  const surveyId = SurveyState.getSurveyId(state)
+  const cycle = SurveyState.getSurveyCycleKey(state)
 
   //assoc current cycle
-  //TODO get current cycle from user prefs
-  props[NodeDef.propKeys.cycles] = [lastSurveyCycle]
+  props[NodeDef.propKeys.cycles] = [cycle]
 
   const nodeDef = NodeDef.newNodeDef(parentUuid, type, props)
```
Comment (minotogna) [inline webapp/survey/nodeDefs/actions.js]: let's pass the cycle to ```NodeDef.newNodeDef```.

File: webapp/survey/nodeDefs/actions.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ import Survey from '../../../common/survey/survey'
 import NodeDef from '../../../common/survey/nodeDef'
 import NodeDefLayout from '../../../common/survey/nodeDefLayout'
 import NodeDefValidations from '../../../common/survey/nodeDefValidations'
+import User from '../../../common/user/user'
```
Comment (minotogna) [inline webapp/survey/nodeDefs/actions.js]: is User still used?

Comment (fefrain) [reply webapp/survey/nodeDefs/actions.js]: no!

**Pull request merged**